indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/RecordSupplierInputSource.java

```diff
@@ -68,16 +70,20 @@ public RecordSupplierInputSource(
     this.recordSupplier = recordSupplier;
     this.useEarliestOffset = useEarliestOffset;
     this.iteratorTimeoutMs = iteratorTimeoutMs;
+
     try {
       assignAndSeek(recordSupplier);
     }
     catch (InterruptedException e) {
-      throw new SamplerException(e, "Exception while seeking to partitions");
+      throw new SamplerException(e, "Thread interrupted while seeking to partitions");
+    }
+    catch (StreamException e) {
```

Comment:
I think I might prefer your alternative approach of catching all `Exception`s here and wrapping them up in a `SamplerException`, as we are not doing anything special with a `StreamException`.

Reply:
Done.